src/Dialogs/Device/DeviceEditWidget.cpp

```diff
@@ -396,8 +418,11 @@ DeviceEditWidget::Save(bool &_changed) noexcept
 
   changed |= FinishPortField(config, (const DataFieldEnum &)GetDataField(Port));
 
-  if (config.MaybeBluetooth())
+  changed |= SaveValueEnum(EngineTypes, config.engine_type);
```

Comment:
Did you mean to move this inside the `MaybeBluetooth` block?
(Also, please add a space between `){`)

Reply:
Okay something is wrong. If I move ` changed |= SaveValueEnum(EngineTypes, config.engine_type);` into the `if (config.MaybeBluetooth())` block, nothing happens. There is no change of engine_type.